internal/hammer/hammer.go

```diff
@@ -443,20 +448,27 @@ func (t *Throttle) Run(ctx context.Context) {
 		case <-ctx.Done(): //context cancelled
 			return
 		case <-ticker.C:
-			tokenCount := t.opsPerSecond
-			timeout := time.After(interval)
-		Loop:
-			for i := 0; i < t.opsPerSecond; i++ {
-				select {
-				case t.tokenChan <- true:
-					tokenCount--
-				case <-timeout:
-					break Loop
-				}
-			}
+			ctx, cancel := context.WithTimeout(ctx, interval)
+			t.supplyTokens(ctx)
+			cancel()
+		}
+	}
+}
+
+func (t *Throttle) supplyTokens(ctx context.Context) {
+	t.mu.Lock()
```

Comment:
Ok, so worst case the UI gets blocked for `interval` while the lock is held here and we're sleeping in the `select` because the channel is full? Which seems fine.

Reply:
Yeah that's exactly right. I tested this by setting a large throttle value and no consumers and it makes the UI feel laggy in that situation, but it works. If we wanted to fix that we could make this throttle more complicated to allow a pending update message to be left, but that's out of scope for now.